test/syncenginetestutils.h

```diff
@@ -656,7 +658,7 @@ class FakeFolder
         return rootTemplate;
     }
 
-    FileInfo currentRemoteState() { return _fakeQnam->currentRemoteState(); }
+    FileInfo &currentRemoteState() { return _fakeQnam->currentRemoteState(); }
```

Comment:
The point of having it named "current" is that it was assured to return a copy. It's also not a "state" if you modify it.

To be clean I think you need to add a setter in FileModifier, as this is what the interface is for.